tests/cel/snippetsFilter_test.go

```diff
@@ -0,0 +1,197 @@
+
+	ngfAPIv1alpha1 "github.com/nginx/nginx-gateway-fabric/v2/apis/v1alpha1"
+)
+
+func TestSnippetsFilterValidation_Snippets(t *testing.T) {
+	t.Parallel()
+	k8sClient := getKubernetesClient(t)
+
+	tests := []struct {
+		name       string
+		wantErrors []string
+		spec       ngfAPIv1alpha1.SnippetsFilterSpec
+	}{
+		{
+			name: "Validate single snippet with valid context",
+			spec: ngfAPIv1alpha1.SnippetsFilterSpec{
+				Snippets: []ngfAPIv1alpha1.Snippet{
+					{
+						Context: ngfAPIv1alpha1.NginxContextHTTP,
+						Value:   "limit_req zone=one burst=5 nodelay;",
+					},
+				},
+			},
+		},
+		{
+			name: "Validate multiple snippets with unique contexts",
+			spec: ngfAPIv1alpha1.SnippetsFilterSpec{
+				Snippets: []ngfAPIv1alpha1.Snippet{
+					{
+						Context: ngfAPIv1alpha1.NginxContextMain,
+						Value:   "worker_processes 4;",
+					},
+					{
+						Context: ngfAPIv1alpha1.NginxContextHTTPServer,
+						Value:   "server_name example.com;",
+					},
+				},
+			},
+		},
+		{
+			name:       "Validate empty snippet value is not allowed",
+			wantErrors: []string{"Invalid value: \"\""},
+			spec: ngfAPIv1alpha1.SnippetsFilterSpec{
+				Snippets: []ngfAPIv1alpha1.Snippet{
+					{
+						Context: ngfAPIv1alpha1.NginxContextHTTP,
+						Value:   "",
+					},
+				},
+			},
+		},
+		{
+			name:       "Validate no snippets is not allowed",
+			wantErrors: []string{"Invalid value: 0"},
+			spec: ngfAPIv1alpha1.SnippetsFilterSpec{
+				Snippets: []ngfAPIv1alpha1.Snippet{},
+			},
+		},
+		{
+			name: "Validate maximum number of snippets is allowed",
+			spec: ngfAPIv1alpha1.SnippetsFilterSpec{
+				Snippets: []ngfAPIv1alpha1.Snippet{
+					{
+						Context: ngfAPIv1alpha1.NginxContextMain,
+						Value:   "worker_processes 4;",
+					},
+					{
+						Context: ngfAPIv1alpha1.NginxContextHTTP,
+						Value:   "limit_req zone=one burst=5 nodelay;",
+					},
+					{
+						Context: ngfAPIv1alpha1.NginxContextHTTPServer,
+						Value:   "server_name example.com;",
+					},
+					{
+						Context: ngfAPIv1alpha1.NginxContextHTTPServerLocation,
+						Value:   "deny all;",
+					},
+				},
+			},
+		},
+		{
+			name:       "Validate more than maximum number of snippets is not allowed",
+			wantErrors: []string{"Too many: 5: must have at most 4 items"},
+			spec: ngfAPIv1alpha1.SnippetsFilterSpec{
+				Snippets: []ngfAPIv1alpha1.Snippet{
+					{
+						Context: ngfAPIv1alpha1.NginxContextMain,
+						Value:   "worker_processes 4;",
+					},
+					{
+						Context: ngfAPIv1alpha1.NginxContextHTTP,
+						Value:   "limit_req zone=one burst=5 nodelay;",
+					},
+					{
+						Context: ngfAPIv1alpha1.NginxContextHTTPServer,
+						Value:   "server_name example.com;",
+					},
+					{
+						Context: ngfAPIv1alpha1.NginxContextHTTPServerLocation,
+						Value:   "deny all;",
+					},
+					{
+						Context: ngfAPIv1alpha1.NginxContextHTTPServerLocation,
+						Value:   "extra configuration",
+					},
+				},
+			},
+		},
+	}
```

Comment:
These are not CEL checks, so we don't need to test them here. The only CEL check we have is that only snippet is allowed per context.